sdformat_urdf/test/joint_tests.cpp

```diff
@@ -0,0 +1,251 @@
+  EXPECT_FALSE(errors.empty());
+  EXPECT_NO_ALGORITHM_ERRORS(errors);
+  ASSERT_FALSE(model);
+}
+
+TEST(Joint, joint_prismatic)
+{
+  sdf::Errors errors;
+  urdf::ModelInterfaceSharedPtr model = sdformat_urdf::parse(
+    get_file(PATH_TO_SDF_JOINT_PRISMATIC), errors);
+  EXPECT_TRUE(errors.empty()) << errors;
+  ASSERT_TRUE(model);
+  ASSERT_EQ("joint_prismatic", model->getName());
+
+  urdf::JointConstSharedPtr joint = model->getJoint("joint_prismatic");
+
+  EXPECT_EQ("joint_prismatic", joint->name);
+  EXPECT_EQ(urdf::Joint::PRISMATIC, joint->type);
+  EXPECT_DOUBLE_EQ(0, joint->axis.x);
+  EXPECT_DOUBLE_EQ(1, joint->axis.y);
+  EXPECT_DOUBLE_EQ(0, joint->axis.z);
+  ASSERT_NE(nullptr, joint->dynamics);
+  EXPECT_DOUBLE_EQ(0, joint->dynamics->damping);
+  EXPECT_DOUBLE_EQ(0, joint->dynamics->friction);
+  ASSERT_NE(nullptr, joint->limits);
+  EXPECT_DOUBLE_EQ(-0.2, joint->limits->lower);
+  EXPECT_DOUBLE_EQ(0.2, joint->limits->upper);
+  EXPECT_DOUBLE_EQ(-1, joint->limits->effort);  // SDFormat default
+  EXPECT_DOUBLE_EQ(-1, joint->limits->velocity);  // SDFormat default
+  ASSERT_EQ(nullptr, joint->safety);
+  ASSERT_EQ(nullptr, joint->calibration);
+  ASSERT_EQ(nullptr, joint->mimic);
+}
+
+TEST(Joint, joint_revolute)
+{
+  sdf::Errors errors;
+  urdf::ModelInterfaceSharedPtr model = sdformat_urdf::parse(
+    get_file(PATH_TO_SDF_JOINT_REVOLUTE), errors);
+  EXPECT_TRUE(errors.empty()) << errors;
+  ASSERT_TRUE(model);
+  ASSERT_EQ("joint_revolute", model->getName());
+
+  urdf::JointConstSharedPtr joint = model->getJoint("joint_revolute");
+
+  EXPECT_EQ("joint_revolute", joint->name);
+  EXPECT_EQ(urdf::Joint::REVOLUTE, joint->type);
+  ASSERT_NE(nullptr, joint->dynamics);
+  EXPECT_DOUBLE_EQ(0, joint->dynamics->damping);
+  EXPECT_DOUBLE_EQ(0, joint->dynamics->friction);
+  ASSERT_NE(nullptr, joint->limits);
+  EXPECT_DOUBLE_EQ(-1.5, joint->limits->lower);
+  EXPECT_DOUBLE_EQ(1.5, joint->limits->upper);
+  EXPECT_DOUBLE_EQ(-1, joint->limits->effort);  // SDFormat default
+  EXPECT_DOUBLE_EQ(-1, joint->limits->velocity);  // SDFormat default
+  ASSERT_EQ(nullptr, joint->safety);
+  ASSERT_EQ(nullptr, joint->calibration);
+  ASSERT_EQ(nullptr, joint->mimic);
+}
+
+TEST(Joint, joint_revolute2)
+{
+  sdf::Errors errors;
+  urdf::ModelInterfaceSharedPtr model = sdformat_urdf::parse(
+    get_file(PATH_TO_SDF_JOINT_REVOLUTE2), errors);
+  EXPECT_FALSE(errors.empty());
+  EXPECT_NO_ALGORITHM_ERRORS(errors);
+  ASSERT_FALSE(model);
+}
+
+TEST(Joint, joint_revolute_axis)
+{
+  sdf::Errors errors;
+  urdf::ModelInterfaceSharedPtr model = sdformat_urdf::parse(
+    get_file(PATH_TO_SDF_JOINT_REVOLUTE_AXIS), errors);
+  EXPECT_TRUE(errors.empty()) << errors;
+  ASSERT_TRUE(model);
+  ASSERT_EQ("joint_revolute_axis", model->getName());
+
+  urdf::JointConstSharedPtr joint = model->getJoint("joint_revolute");
+
+  EXPECT_EQ("joint_revolute", joint->name);
+  EXPECT_EQ(urdf::Joint::REVOLUTE, joint->type);
+  EXPECT_DOUBLE_EQ(0.1, joint->axis.x);
```

Comment:
In libsdformat10 (https://github.com/osrf/sdformat/pull/312), the joint axis will be normalized. Consider changing the test and the corresponding SDFormat file to use unit vector so we don't have to update this when we move to libsdformat10.

Reply:
Compares normalized joint axes in 84d23f5